src/solver/optimisation/post_process_commands.cpp

```diff
@@ -209,8 +213,97 @@ void CurtailmentSharingPostProcessCmd::execute(const struct optRuntimeData& opt_
     unsigned int week = opt_runtime_data.week;
 
     // TO DO : move the curtailment sharing post process here
+    double totalLmrViolation = calculateDensNewAndTotalLmrViolation();
+    logs.info() << "[adq-patch] Year:" << year + 1 << " Week:" << week + 1
+        << ".Total LMR violation:" << totalLmrViolation;
+    const std::set<int> hoursRequiringCurtailmentSharing = getHoursRequiringCurtailmentSharing();
+    HourlyCSRProblem hourlyCsrProblem(problemeHebdo_);
+    for (int hourInWeek : hoursRequiringCurtailmentSharing)
+    {
+        logs.info() << "[adq-patch] CSR triggered for Year:" << year + 1
+            << " Hour:" << week * nbHoursInWeek + hourInWeek + 1;
+        hourlyCsrProblem.setHour(hourInWeek);
+        hourlyCsrProblem.run(week, year);
+    }
+
+}
+
+double CurtailmentSharingPostProcessCmd::calculateDensNewAndTotalLmrViolation()
+{
+    double netPositionInit;
+    double densNew;
+    double totalNodeBalance;
+    double totalLmrViolation = 0.0;
+    const int numOfHoursInWeek = 168;
+
+    for (int Area = 0; Area < problemeHebdo_->NombreDePays; Area++)
+    {
+        if (problemeHebdo_->adequacyPatchRuntimeData.areaMode[Area] == physicalAreaInsideAdqPatch)
+        {
+            for (int hour = 0; hour < numOfHoursInWeek; hour++)
+            {
+                std::tie(netPositionInit, densNew, totalNodeBalance) = calculateAreaFlowBalance(problemeHebdo_, Area, hour);
+
+                // adjust densNew according to the new specification/request by ELIA
+                /* DENS_new (node A) = max [ 0; ENS_init (node A) + net_position_init (node A)
+                                        + ? flows (node 1 -> node A) - DTG.MRG(node A)] */
+                auto& scratchpad = *(area_list_[Area]->scratchpad[thread_number_]);
```

Comment:
There was a `const` here. Where'd it go ?

Reply:
Where was the **const** you're talking about ?

Reply:
done